doc/rpc.rst

```diff
@@ -0,0 +1,54 @@
+.. _rpc:
+
+********
+Dune RPC
+********
+
+Starting from dune 3.0, dune's watch mode also runs an RPC server. This is a
+general mechanism introduced to integrate with various dune functionality. Some
+use cases we have in mind are:
+
+- Text editors that would like to receive dune's error reporting
+- Custom actions that need to communicate with dune
+
+There's no fixed scope for RPC, and we encourage users to submit requests to
+cover more functionality.
+
+The purpose of this documentation is to explain how RPC works, and how to
+connect to it as a client. More concrete information such as what requests are
+available is in `dune_rpc.mli`
+
+`dune-rpc` library
+==================
+
+We provide a client library ``dune-rpc`` to make it easy to write clients. The
+library has a versioned interface and we guarantee to maintain its stability.
+With that being said, we reserve te right to:
+
+- Add optional arguments to functions
+- Add new fields to records
+- New variant constructors that will not cause runtime errors in existing user
+  programs.
```

Comment:
I would expect this to go in the mli file. Could go in both, but it should go at least in the mli.

BTW, regarding:

> - Add new fields to records

such records should be `private` then. Otherwise it's a too easy mistake to make.

Reply:
I'll save changing them to private in another PR

Reply:
Ack. BTW, I'd still write the code in a fully defensive way and make all types abstract, rather than rely on a convention written in the doc. I'm guessing you followed @bobot's idea of library declaring conventions for backward compatibility?

While I like this idea, part of the idea is to have tooling to verify that the contract is being respected. Without such automatic verification, it feels a bit too fragile to me and I'd stick to the proven bullet-proof way of using abstract types.